theseus/core/vectorizer.py

```diff
@@ -328,16 +360,35 @@ def _handle_schema_vectorization(
             len(cost_fn_wrappers),
         )
         Vectorize._compute_error_and_replace_wrapper_caches(
-            vectorized_cost_fn, cost_fn_wrappers, batch_size
+            vectorized_cost_fn, cost_fn_wrappers, batch_size, mode
         )
 
-    def _vectorize(self):
-        self._clear_wrapper_caches()
-        for schema, cost_fn_wrappers in self._schema_dict.items():
+    def _vectorize(
+        self, mode: _VectorizationMode = _VectorizationMode.FULL
```

Comment:
How to select different vectorization mode? According to memory usage?

Reply:
Currently we select it manually in two places:

1.) when requesting to compute `Objective.error()` we select `WEIGHTED_ERROR`, so that we compute the vectorized error w/o modifying the permanent caches. 
2.) when Objective requests that the vectorization is updated, then we use `FULL` so that we compute jacobians and replace the permanent caches. 

It's good that you asked because I just realized that running vectorization for 1.) only makes sense if `error` receives `input_tensors`. But if this is `None` or `empty`, then there is no point in computing vectorization again, we can just return the cached values.

Reply:
Oh nevermind. I was about to fix this, but if we return the jacobians cache we go back to the issue were the error under robust costs is the scaled error instead of the actual error. I guess we could add some logic to prevent recompute if there are no robust costs, but this seems a bit messy and prone to error. I'll leave as is for now.  

FWIW, in code about to be merged in other PRs the optimizers always compute error with `input_tensors` set to the latest values, so I don't think this optimization will save much time.

Reply:
Added.